stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt

```diff
@@ -55,6 +62,51 @@ class ChatFragment : Fragment() {
         initMessagesViewModel()
         initMessageInputViewModel()
         configureBackButtonHandling()
+        configureThreadSubtitle()
+    }
+
+    private fun configureThreadSubtitle() {
+        val subtitleMediator = MediatorLiveData<String?>()
+
+        subtitleMediator.addSource(messageListViewModel.state) { state ->
+            handleSubtitleChange(state, messageListViewModel.channel.value?.name).let(subtitleMediator::setValue)
+        }
+
+        subtitleMediator.addSource(messageListViewModel.channel) { channel ->
+            handleSubtitleChange(messageListViewModel.state.value, channel?.name).let(subtitleMediator::setValue)
+        }
+
+        subtitleMediator.observe(viewLifecycleOwner) { subtitle ->
+            subtitle?.let(binding.messagesHeaderView::setThreadSubtitle)
+        }
+    }
+
+    private fun handleSubtitleChange(state: MessageListViewModel.State?, channelName: String?): String? {
+        return if (state is MessageListViewModel.State.Result && state.messageListItem.isThread) {
```

Comment:
This won't work... a new Message inside a thread can arrive even thought the user is not inside a thread... I need to change this.